src/docfx/build/redirection/RedirectionProvider.cs

```diff
@@ -213,17 +213,21 @@ private static IEnumerable<string> ProbeRedirectionFiles(string docsetPath, Repo
                 var (errors, redirectionSourceMonikers) = _monikerProvider.GetFileLevelMonikers(file);
                 _errorLog.Write(errors);
 
-                List<FilePath> candidates;
+                IEnumerable<FilePath> candidates;
                 if (redirectionSourceMonikers.Length == 0)
                 {
-                    candidates = docs.Where(doc => _monikerProvider.GetFileLevelMonikers(doc).monikers.Length == 0).ToList();
+                    candidates = docs.Where(doc => _monikerProvider.GetFileLevelMonikers(doc).monikers.Length == 0);
                 }
                 else
                 {
-                    candidates = docs.Where(doc => _monikerProvider.GetFileLevelMonikers(doc).monikers.Intersect(redirectionSourceMonikers).Any()).ToList();
+                    candidates = docs.Where(doc => _monikerProvider.GetFileLevelMonikers(doc).monikers.Intersect(redirectionSourceMonikers).Any());
+                }
+
+                if (candidates.Any())
```

Comment:
This will call the enumerator twice, once in `Any`, once in `OrderBy`

Reply:
`Any` will execute only 1 iteration, but still `Orderby` and the later `foreach` will execute the `enumeration` twice